app/services/description_import.rb

```diff
@@ -89,4 +95,36 @@ def compact_params(params)
       params[key].map { |param| compact_params(param) }
     end
   end
+
+  def remove_contributors_if_role_without_name(compacted_params_hash)
+    return unless compacted_params_hash && compacted_params_hash[:contributor]
+
+    compacted_params_hash[:contributor].each do |contributor|
+      next unless contributor && contributor[:name].nil? && contributor[:role].present?
+
+      compacted_params_hash[:contributor].delete(contributor)
+    end
+  end
+
+  def remove_form_if_source_without_value(compacted_params_hash)
+    return unless compacted_params_hash && compacted_params_hash[:form]
+
+    compacted_params_hash[:form].each do |form|
+      next unless form && form[:value].nil? && (form[:source].present? || form[:type].present?)
+
+      compacted_params_hash[:form].delete(form)
+    end
+  end
+
+  def remove_nested_attributes_without_value(compacted_params_hash)
+    # event can have contributors, geographic can have form
+    [:relatedResource, :event, :geographic].each do |parent_property|
+      next if compacted_params_hash[parent_property].blank?
```

Comment:
Are we protecting against `nil`s or empty arrays or both or something else here?

Reply:
we only care to process if there is a (parent_property) present in the hash.  It could also be worded:

```next unless compacted_params_hash[parent_property].present?``` 

and I had it that way, but it seemed more confusing.

Yes - we are protecting against nil and also empty arrays.  It all depends on what columns are in the spreadsheet, at this point.